openapi.yaml

```diff
@@ -26,7 +28,552 @@ tags:
+      operationId: ListUsers
+      description: |
+        List all users
+      responses:
+        '200':
+          description: Array of registered users
+    post:
+      tags:
+        - admins
+        - User
+      summary: Create a new User
+      operationId: CreateUser
+      description: |
+        Create a new User
+      requestBody:
+        description: User information
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/CreateUserRequest'
+      responses:
+        '200':
+          description: User added
+  /api/users/bulk:
+    post:
+      tags:
+        - admins
+        - User
+      summary: Create users in bulk
+      operationId: CreateUsers
+      description: |
+        Create a new User
+      requestBody:
+        description: Bulk User information
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/CreateUsersRequest'
+      responses:
+        '200':
+          description: Users added
+  /api/users/{userId}:
+    parameters:
+      - name: userId
+        in: path
+        required: true
+        schema:
+          $ref: '#/components/schemas/UserId'
+    put:
+      tags:
+        - admins
+        - User
+      summary: Update user
+      operationId: UpdateUser
+      requestBody:
+        description: User information
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/UpdateUserRequest'
+      description: |
+        Update a user
+      responses:
+        '200':
+          description: User updated
+    delete:
+      tags:
+        - admins
+        - User
+      summary: Delete a User
+      operationId: DeleteUser
+      description: |
+        Delete a User
+      responses:
+        '200':
+          description: User deleted
+  /api/users/{userId}/password:
+    put:
+      tags:
+        - admins
+        - User
+      summary: Update user password
+      operationId: UpdateUserPassword
+      parameters:
+        - name: userId
+          in: path
+          required: true
+          schema:
+            $ref: '#/components/schemas/UserId'
+      requestBody:
+        description: password of the user
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/UpdateUserPasswordRequest'
+      description: |
+        Update a user
+      responses:
+        '200':
+          description: User password updated
+  /api/user-roles:
+    get:
+      tags:
+        - admins
+        - guests
+        - researchers
+        - UserRoles
+      summary: Fetch all user roles present in Service Workbench
+      operationId: ListUserRoles
+      description: |
+        Fetch all user roles present in Service Workbench
+      responses:
+        '200':
+          description: Array of all the registered user roles
+  /api/template/{templateId}:
+    get:
+      tags:
+        - admins
```

Comment:
Is this an endpoint used in BYOB as well?

Reply:
For BYOB we are using this API data-sources/accounts/:awsAccountId/cfn This one isn't used for BYOB